### PR TITLE
chore: valid title

### LINKED ISSUE #247: valid-title should allow functions
As of https://github.com/vitest-dev/vitest/issues/3493 -> https://github.com/vitest-dev/vitest/pull/3497, Vitest `describe`s & similar can take in function names. I think the `valid-title` rule should by default no longer complain if a function name is provided.

Before:

```ts
import { describe } from "vitest"

function isBodyWithReplies() { /* ... */ }

describe(isBodyWithReplies, () => {
	//   ~~~~~~~~~~~~~~~~~~ Test title must be a string
	// ...
})
```

After:

```ts
import { describe } from "vitest"

function isBodyWithReplies() { /* ... */ }

describe(isBodyWithReplies, () => {
	//   OK
	// ...
});

describe(1234567890, () => {
	//   ~~~~~~~~~~ Test title must be a function or string
	// ...
});
```

I'd be happy to send a PR for this! ❤️

### PR DESCRIPTION
**Pull request by veritem**
Fixes #247 #251

### CODE REVIEW

File: docs/rules/valid-title.md [hunk 1]
```diff
@@ -168,9 +168,34 @@ Examples of **correct** code for this rule with the `{ "mustMatch": { "describe"
...
+  // This check succeeds because the title matches the regex
   it('should be a number.', () => {
     expect(1).toBeNumber()
   })
 })
 ```
+
+Note: If you'd like to use a function or class names inside `describe`, `test` or `it` blocks as a parameter, you must enable vitest's type checking.
+
+To enable typechecking for vistest make sure settings key is added in your configuration
```
Comment (so1ve) [inline docs/rules/valid-title.md:180]: Typo?

Comment (veritem) [reply docs/rules/valid-title.md:180]: @so1ve, how do you want it to be phrased?

Comment (so1ve) [reply docs/rules/valid-title.md:180]: I mean `vistest`

Comment (veritem) [reply docs/rules/valid-title.md:180]: ooh good catch thanks!

**Pull request merged**